/**
 * Copyright 2017 Erik Zenker
 *
 * This file is part of Graybat.
 *
 * Graybat is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Graybat is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with Graybat.
 * If not, see <http://www.gnu.org/licenses/>.
 */

syntax = "proto3";

service Signaling {
    rpc  RequestContext(ContextRequest) returns (ContextReply) {}
    rpc  RequestVaddr(VaddrRequest) returns (VaddrReply) {}
    rpc  LookupVaddr(VaddrLookup) returns (UriReply){}
    rpc  LeaveContext(LeaveRequest) returns (LeaveReply){}
}

message ContextRequest {
    string context_name = 1;
}

message LeaveRequest {
    string context_name = 1;
}

message LeaveReply {
}

message VaddrRequest {
    sint64 context_id = 1;
    string data_uri = 2;
    string ctrl_uri = 3;
}

message VaddrLookup {
    sint64 context_id = 1;
    sint64 vaddr = 2;
}

message ContextReply {
    sint64 context_id = 1;
}

message VaddrReply {
    sint64 vaddr = 1;
}

message UriReply {
    string data_uri = 1;
    string ctrl_uri = 2;
}